src/NServiceBus.Core/Routing/Routers/UnicastSendRouterConnector.cs

```diff
@@ -41,7 +45,10 @@ public enum RouteOption
             {
                 throw new InvalidOperationException("Cannot route to a specific instance because an endpoint instance discriminator was not configured for the destination endpoint. It can be specified via EndpointConfiguration.MakeInstanceUniquelyAddressable(string discriminator).");
             }
-            var thisEndpoint = state.Option == RouteOption.RouteToAnyInstanceOfThisEndpoint ? sharedQueue : null;
+
+            var effectiveSharedQueue = ApplyDistributorLogic(context);
```

Comment:
nitpicking: I know what you mean with that variable name, still I think effective is the wrong description here.